test/variant-handler.test.ts

```diff
@@ -0,0 +1,91 @@
+import { createGenerator } from '@unocss/core'
+import { describe, expect, test } from 'vitest'
+
+describe('variants', () => {
+  test('variant context is propagated', async () => {
+    const uno = createGenerator({
+      rules: [
+        ['foo', { name: 'bar' }],
+      ],
+      variants: [
+        {
+          match: (input: string) => {
```

Comment:
```suggestion
          match(input) {
```

I guess the type can be inferred